quilt/tools/build.py

```diff
@@ -144,29 +172,49 @@ def _callback(count):
 
     return df
 
-def build_package(username, package, yaml_path):
+def build_package(username, package, yaml_path, checks_path=None, dryrun=False, env='default'):
     """
     Builds a package from a given Yaml file and installs it locally.
 
     Returns the name of the package.
     """
+    def find(key, value):
+        for k, v in value.items():
+            if k == key:
+                yield v
+            elif isinstance(v, dict):
+                for result in find(key, v):
+                    yield result
+            elif isinstance(v, list):
+                for d in v:
+                    for result in find(key, d):
+                        yield result
     build_dir = os.path.dirname(yaml_path)
     fd = open(yaml_path)
     docs = yaml.load_all(fd)
-    data = next(docs, None) # leave other dicts in the generator
-    if not isinstance(data, dict):
-        raise BuildException("Unable to parse YAML: %s" % yaml_path)
-
-    build_package_from_contents(username, package, build_dir, data)
-
-def build_package_from_contents(username, package, build_dir, data):
-    """
-    Builds package according to YAML node
-    """
-    contents = data.get('contents', {})
+    build_data = next(docs, None) # leave other dicts in the generator
+    if not isinstance(build_data, dict):
+        raise BuildException("Unable to parse build YAML: %s" % yaml_path)
+    if len(list(find('checks', build_data['contents']))) > 0 and 'checks' not in build_data:
+        checks_path = 'checks.yml'
+
+    checks_contents = None
+    if checks_path is not None:
+        fd = open(checks_path)
+        docs = yaml.load_all(fd)
+        contents = next(docs, None)['contents'] # leave other dicts in the generator
+        if not isinstance(contents, dict):
```

Comment:
There might be all kinds of other errors, too - e.g., `next` could return `None` or `contents` might be completely missing. (You can use `.get('contents')` to deal with that).

Unfortunately, there doesn't seem to be an easy way to validate yaml.

Reply:
???  sorry, I'm unclear what you want?